src/core/p5.Renderer2D.js

```diff
@@ -114,18 +114,31 @@ p5.Renderer2D.prototype._getTintedImageCanvas = function (img) {
   this._tintCanvas.height = img.canvas.height;
   var tmpCtx = this._tintCanvas.getContext('2d');
 
-  // this._tint stores rgba values on scale form 0-255. To set fillStyle with
-  // 'rgba', the alpha needs to be on a 0 to 1 scale.
-  var rgba = this._tint.slice(0,3).join(',') + ', ' + this._tint[3] / 255;
-  tmpCtx.fillStyle = 'rgba(' + rgba + ')';
-  tmpCtx.fillRect(0, 0, this._tintCanvas.width, this._tintCanvas.height);
-  tmpCtx.globalCompositeOperation = 'destination-atop';
-  tmpCtx.drawImage(img.canvas, 0, 0, this._tintCanvas.width,
-    this._tintCanvas.height);
-  tmpCtx.globalCompositeOperation = 'multiply';
-  tmpCtx.drawImage(img.canvas, 0, 0, this._tintCanvas.width,
-    this._tintCanvas.height);
-  return this._tintCanvas;
+  // Special case of 'white tint' which affects only alpha - default to
+  // P5.prototype tint. Otherwise, use code-dot-org specific behavior where tint
+  // only affects the RGB
+  // of the sprite and the alpha of tint changes the 'strength' of the tint.
+  if (this._alpha) {
+    // Set the p5 renderer tint to the current tint value.
+    // Alpha stored as value between 0 to 1, but tint is stored as value between
+    // 0 and 255.
+    p5.prototype._renderer = {};
+    p5.prototype._renderer._tint = [255, 255, 255, this._alpha * 255];
+    return p5.prototype._getTintedImageCanvas(img);
```

Comment:
super nit: since this conditional returns, we could simplify nesting a little bit here:

```javascript
if (this._alpha) {
  // new code
  return;
}

// existing code
```